ci/ray_ci/tester_container.py

```diff
@@ -43,20 +45,33 @@ def run_tests(
         """
         Run tests parallelly in docker.  Return whether all tests pass.
         """
-        chunks = [
-            shard_tests(test_targets, self.shard_count, i) for i in self.shard_ids
-        ]
+        # shard tests and remove empty chunks
+        chunks = list(
+            filter(
+                len,
+                [
+                    shard_tests(test_targets, self.shard_count, i)
+                    for i in self.shard_ids
+                ],
+            )
+        )
+        if not chunks:
+            # no tests to run
+            return True
+
+        # divide gpus evenly among chunks
+        gpu_ids = chunk_into_n(list(range(self.gpus)), len(chunks))
```

Comment:
are you sure each one will have at least 1 (or some minimal amount)? should we add some unit tests for this?

Reply:
gpu might become 0 if the users go crazy yeah, but I'll leave it up to the user to compute the resource correctly (through number of gpus and number of parallelization flag). As long as gpus <= parallelization this should be be non-zero, but then users also need to make sure each test can run with floor(gpus/parallelization) core.

Reply:
hmm.. maybe consider provide a clearer error message? like if the user go crazy (for whatever reason) and can't dig themselves out, they will just ping you anyways.

Reply:
good point, the constraint can probably be as simple as gpus <= parallelization; as long as that is true, gpu should be greater than 0